src/components/radiobutton/index.tsx

```diff
@@ -30,10 +30,11 @@ export const Radio = React.forwardRef<HTMLLabelElement, RadioButtonProps>(
         <label className={cx('label', disabled && 'labelDisabled', className)} htmlFor={id || defaultId}>
           <input
             type="radio"
-            checked={checked}
             className={cx('radiobox', className)}
             id={id || defaultId}
             disabled={disabled}
+            checked={value === id}
+            value={id}
```

Comment:
if we want to control the state outside the component.
I think add smth like `checked={checked || value === id}` and `value={value || id || defaultId}`